### PR TITLE
Use libcoarse in the file and the devdax providers

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Use libcoarse in the file and the devdax providers.

This PR is required by 3 PRs that fix the following issues:
- https://github.com/oneapi-src/unified-memory-framework/issues/864
- https://github.com/oneapi-src/unified-memory-framework/issues/900
- https://github.com/oneapi-src/unified-memory-framework/issues/904

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
This PR is required by 3 PRs that fix the following issues:
- https://github.com/oneapi-src/unified-memory-framework/issues/864
- https://github.com/oneapi-src/unified-memory-framework/issues/900
- https://github.com/oneapi-src/unified-memory-framework/issues/904

@vinser52 @bratpiorka please review

File: src/provider/provider_devdax_memory.c [hunk 1]
```diff
@@ -189,23 +214,25 @@ static umf_result_t devdax_initialize(void *params, void **provider) {
...
-        }
-
-        goto err_free_devdax_provider;
+        goto err_unmap_devdax;
     }
 
     LOG_DEBUG("devdax memory mapped (path=%s, size=%zu, addr=%p)",
               in_params->path, devdax_provider->size, devdax_provider->base);
 
+    ret = coarse_add_memory_fixed(coarse, devdax_provider->base,
```
Comment (bratpiorka) [inline src/provider/provider_devdax_memory.c:229]: please comment that you are adding the entire devdax memory here as a single block

File: src/provider/provider_devdax_memory.c [hunk 2]
```diff
@@ -161,21 +169,38 @@ static umf_result_t devdax_initialize(void *params, void **provider) {
 
     memset(devdax_provider, 0, sizeof(*devdax_provider));
 
-    ret = devdax_translate_params(in_params, devdax_provider);
+    coarse_params_t coarse_params = {0};
+    coarse_params.provider = devdax_provider;
+    coarse_params.page_size = DEVDAX_PAGE_SIZE_2MB;
+    coarse_params.cb.alloc = NULL; // not available for the fixed-size provider
```
Comment (bratpiorka) [inline src/provider/provider_devdax_memory.c:175]: fixed-size -> devdax?

**Review by bratpiorka — APPROVED** [review]

File: src/provider/provider_devdax_memory.c (hunk 1 above)
Comment (ldorau) [reply src/provider/provider_devdax_memory.c:229]: Done

File: src/provider/provider_devdax_memory.c (hunk 2 above)
Comment (ldorau) [reply src/provider/provider_devdax_memory.c]: Done

**Comment by ldorau** [comment]
@vinser52 @lplewa @lukaszstolarczuk @PatKamin @kswiecicki @KFilipek please review

**Comment by ldorau** [comment]
Please review

**Review by kswiecicki — APPROVED** [review]

**Pull request merged**